### PR TITLE
Build PCRE2 on Windows

### PR DESCRIPTION
**Pull request by HertzDevil**
Builds `pcre2-8.lib` and distributes it alongside `pcre.lib` for Windows packages.

~~This makes Crystal always prefer PCRE2 on Windows by default, because there is no `pkg-config` here.~~

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

**Comment by HertzDevil** [comment]
@straight-shoota feel free to include this patch as part of the new PCRE2 PR directly

**Comment by straight-shoota** [comment]
I rebased this on top of the new PR for PCRE2 (#12856). But it doesn't work as is.

In #12856 PCRE2 is not chosen by default. It's opt-in via `-Duse_pcre2` for now. And for now we also shouldn't build the compiler with PCRE2. We're not doing that on other platforms yet either, and it affects the syntax of the language (#12857).

We could run regex specs against PCRE2 on windows. But I suppose the specs from #12856 should be sufficient for that. I don't expect different behaviour from PCRE2 on other platforms.

File: .github/workflows/win.yml [hunk 1]
```diff
@@ -318,7 +332,7 @@ jobs:
         run: make -f Makefile.win deps
       - name: Link Crystal executable
         run: |
-          Invoke-Expression "cl crystal.obj /Fecrystal src\llvm\ext\llvm_ext.obj $(llvm\bin\llvm-config.exe --libs) libs\pcre.lib libs\gc.lib WS2_32.lib advapi32.lib libcmt.lib dbghelp.lib ole32.lib shell32.lib legacy_stdio_definitions.lib /link /LIBPATH:$(pwd)\libs /STACK:0x800000 /ENTRY:wmainCRTStartup"
+          Invoke-Expression "cl crystal.obj /Fecrystal src\llvm\ext\llvm_ext.obj $(llvm\bin\llvm-config.exe --libs) libs\pcre2-8.lib libs\gc.lib WS2_32.lib advapi32.lib libcmt.lib dbghelp.lib ole32.lib shell32.lib legacy_stdio_definitions.lib /link /LIBPATH:$(pwd)\libs /STACK:0x800000 /ENTRY:wmainCRTStartup"
```
Comment (HertzDevil) [inline .github/workflows/win.yml:335]: First link should probably use the old lib too

**Comment by straight-shoota** [comment]
Looks like this was claused automatically by merging #12856. No idea why. The target branch should've just rebased (as did https://github.com/crystal-lang/crystal/pull/12866#event-8098963789).

**Comment by oprypin** [comment]
Yeah GitHub behaves weirdly.
Now what you can do it undelete the branch then manually change the base here

**Comment by straight-shoota** [comment]
This is so weird. But it works 🤷 Thanks @oprypin 👍

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**